R/peakdet.R

```diff
@@ -1,109 +1,112 @@
-#' Find_true_occurence
+#' Helper function to find the borders of a peak in a sequence of intensities
 #'
-#' Loads a .xlsx or .csv file that contains all the information needed to detect targeted compounds.
+#' @param sign_change `logical` with positions of local minima and maxima.
+#'     Length of `sign_change` needs to be the same than the length of the
+#'     original intensity vector. Eventually add leading and trailing `FALSE`.
 #'
-#' @param signs
-#' @param vector
-#' @param point
-
-#' @return
-#' @export
+#' @param index `integer(1)` with the index of the detected peak
 #'
-
-
-find_true_occurrence <- function(signs,vector,point) {
-  left_true <- NA
-  right_true <- NA
-
-  if(point == 1){
-    point = point + 1
-  }
-
-
-
-  # Search for TRUE to the left --> if sign changes and at least doens't change at the next point
-  for (i in (point - 1):1) {
-    if (signs[i] == TRUE &&  point - i > 2) {
-      left_true <- i
-      break
-    } else{ left_true <- 1}
-  }
-
-  if(point >= length(signs)){
-    point = length(signs) -1
-  }
-
-  # Search for TRUE to the right
-  for (i in (point + 1):length(signs)) {
-    if (signs[i] == TRUE && i - point > 2) {
-      right_true <- i
-      break
-    } else{ right_true <- length(vector)}
-  }
-
-  result <- list(left_true = left_true, right_true = right_true)
-  return(result)
+#' @param min_dist `integer(1)` defining the minimum distance of the left and
+#'     right border to the apex position. With `min_dist = 2` (the default)
+#'     the right border needs to be >= `index + 2`
+#'
+#' @return named `numeric(2)` with the left and right index of the border. If
+#'     no local minima were found 1 and `length(sign_change)` are returned.
+#' 
+#' @noRd
+.find_peak_border <- function(sign_change, index, min_dist = 2) {
```

Comment:
Changed that to an internal function. To get the exact same results as in the original version parameter `min_dist` should be set to `min_dist = 3`. See comment above.